pkg/pipeline/run.go

```diff
@@ -53,32 +53,27 @@ func Run(pc *config.Provider, rootDir string) { // nolint:gocyclo
 	}
 
 	// Add ProviderConfig API package to the list of API version packages.
-	apiVersionPkgList := make([]string, 0)
+	var apiVersionPkgList []string
 	for _, p := range pc.BasePackages.APIVersion {
 		apiVersionPkgList = append(apiVersionPkgList, filepath.Join(pc.ModulePath, p))
 	}
 	// Add ProviderConfig controller package to the list of controller packages.
-	controllerPkgList := make([]string, 0)
+	var controllerPkgList []string
 	for _, p := range pc.BasePackages.Controller {
 		controllerPkgList = append(controllerPkgList, filepath.Join(pc.ModulePath, p))
 	}
+
 	count := 0
-	for group, versions := range resourcesGroups {
-		for version, resources := range versions {
+	for _, group := range sortedGroups(resourcesGroups) {
```

Comment:
Is golang clever enough not to call this function in each iteration?

Reply:
Yes https://play.golang.org/p/PNL6sswqke3